python/huggingfaceserver/huggingfaceserver/model.py

```diff
@@ -191,6 +238,84 @@ def load(self) -> bool:
+            if os.path.exists(config_file_path):
+                with open(config_file_path, "r") as f:
+                    config_text = f.read()
+                model_config = ModelConfigGRPC()
+                text_format.Parse(config_text, model_config)
+                config_dict = MessageToDict(model_config)
+                inputs_config = [
+                    TritonTensor(
+                        name=input["name"],
+                        dtype=from_triton_type_to_np_type(input["dataType"]),
+                        shape=input["dims"],
+                        optional=input.get("optional", False),
+                    )
+                    for input in config_dict["input"]
+                ]
+                outputs_config = [
+                    TritonTensor(
+                        name=output["name"],
+                        dtype=from_triton_type_to_np_type(output["dataType"]),
+                        shape=output["dims"],
+                        optional=output.get("optional", False),
+                    )
+                    for output in config_dict["output"]
+                ]
+            else:
+                # Fallback config
+                inputs_config = [
+                    TritonTensor(name="input_ids", dtype=np.int64, shape=(-1, -1)),
+                    TritonTensor(name="attention_mask", dtype=np.int64, shape=(-1, -1)),
+                    TritonTensor(
+                        name="token_type_ids",
+                        dtype=np.int64,
+                        shape=(-1, -1),
+                        optional=True,
+                    ),
+                ]
+                outputs_config = [
+                    TritonTensor(name="outputs", dtype=np.int64, shape=(-1, -1)),
+                ]
+            triton_lifecycle_policy = TritonLifecyclePolicy(
+                launch_triton_on_startup=True, local_model_store=True
+            )
+            self.triton_server = Triton(
+                config=self._triton_config,
+                triton_lifecycle_policy=triton_lifecycle_policy,
+            )
+
+            def infer_fn(requests: List[Request]):
+                responses = []
+                for request in requests:
+                    input_tensors = {}
+                    for input_name, input_array in request.data.items():
+                        input_tensors[input_name] = torch.tensor(
+                            input_array, device=self.device
+                        )
+
+                    if (
+                        self.task == MLTask.text2text_generation.value
+                        or self.task == MLTask.text_generation
+                    ):
+                        outputs = self.model.generate(**input_tensors)
+                    else:
+                        outputs = self.model(**input_tensors)
+                    responses.append({"outputs": outputs.numpy()})
+                return responses
```

Comment:
The inference should be done by triton not calling self.model, I guess `infer_func` is for python backend?